### PR TITLE
sasl-mechs.md: Add ECDSA-NIST256P-CHALLENGE

### PR DESCRIPTION
**Pull request by nyuszika7h**
~~Also, put PLAIN first, because it's the most basic mechanism.~~ I might do that in a separate pull request.

### CODE REVIEW

File: docs/sasl-mechs.md [hunk 1]
```diff
@@ -4,8 +4,9 @@ title: SASL Mechanisms
 ---
 IAL primarily uses the same mechanisms as SASL in other protocols. Most commonly:
 
-* [EXTERNAL](https://tools.ietf.org/html/rfc4422#appendix-A) as defined by RFC 4422
 * [PLAIN](https://tools.ietf.org/search/rfc4616) as defined by RFC 4616
+* [EXTERNAL](https://tools.ietf.org/html/rfc4422#appendix-A) as defined by RFC 4422
+* [ECDSA-NIST256P-CHALLENGE](https://github.com/kaniini/ecdsatool)
```
Comment (Mikaela) [inline docs/sasl-mechs.md]: I think this link should go to https://github.com/kaniini/ecdsatool#mechanism-spec instead and there should be something more formal than it.

Comment (nyuszika7h) [inline docs/sasl-mechs.md]: I don't think so, because you would have to scroll up anyway to see how to use ecdsatool.

**Comment by DanielOaks** [comment]
I agree with putting PLAIN first. Seems to be the globally available one these days.

Should `ECDSA-NIST256P-CHALLENGE` be documented as a formal IRCv3 SASL mech ala [the others](https://github.com/ircv3/ircv3-specifications/tree/master/documentation)? I'm iffy about linking it to a specific tool's readme.

I've never done that much with SASL though, will probably leave for someone else to take a look at.

File: docs/sasl-mechs.md (hunk 1 above)
Comment (Mikaela) [inline docs/sasl-mechs.md]: But ecdsatool isn't related to the specification or do you see those RFCs including anything else either?

**Comment by Mikaela** [comment]
> I agree with putting PLAIN first. Seems to be the most globally available / basic one these days.

I also think that PLAIN and EXTERNAL are good to have as first, I am not sure on Atheme, but with Anope they are hardcoded into the SASL module so you cannot disable them (this was wondered when ZNC removed DH-\* if I recall correctly).

> Should ECDSA-NIST256P-CHALLENGE be documented as a formal IRCv3 SASL mech ala the others? I'm iffy about linking it to a specific tool's readme.

:+1:

**Comment by kaniini** [comment]
:-1:

**Comment by nyuszika7h** [comment]
@kaniini You're not contributing anything to this discussion at all. Is it so hard to explain what you would prefer?

Anyway, I should probably separate the two changes out, though I'm not sure if moving PLAIN is worth a pull request on its own.

File: docs/sasl-mechs.md (hunk 1 above)
Comment (nyuszika7h) [inline docs/sasl-mechs.md]: @Mikaela True, you would probably use a library (or an FFI binding for `libecdsaauth` perhaps) unless you're writing a client in bash.

**Comment by DanielOaks** [comment]
I can throw PLAIN up the top after this PR gets decided.

**Comment by kaniini** [comment]
:-1: to including ECDSA-NIST256P-CHALLENGE at all.  And yes, I get to say that because I made the damned thing.

**Comment by nyuszika7h** [comment]
@kaniini Even if you think it sucks (because of NIST or whatever), it's already being used out in the wild (e.g. freenode), so IMO if DH-AES and DH-BLOWFISH are still mentioned, then ECDSA-NIST256P-CHALLENGE should be listed too. But I haven't heard you or anyone saying that it shouldn't be used, so I don't understand what's the problem.

**Comment by kaniini** [comment]
I don't think what I made sucks, but it is not an IRCv3 standard nor do I have any intentions of submitting it as an IRCv3 standard.  Instead, I plan on revising the standard and introducing a family of mechanisms based on various curve types (NIST, 25519, etc.) through an IETF submission.  But that will require more time than I have to work on it right now.

**Comment by kaniini** [comment]
For what it's worth, documenting DH-AES and DH-BLOWFISH here *is* a mistake because it implies endorsement of those mechanisms, and they do legitimately suck.

**Comment by nyuszika7h** [comment]
> it is not an IRCv3 standard

Nothing on that page implies those are all standard mechanisms. SCRAM-SHA-256 is just a draft as well.

>  Instead, I plan on revising the standard and introducing a family of mechanisms based on various curve types (NIST, 25519, etc.) through an IETF submission.

Nothing is submitted to IETF yet, is it? Using that logic, why not remove everything from the site because nothing is an IETF standard?

**Comment by nyuszika7h** [comment]
> because it implies endorsement of those mechanisms

There's a big red warning.

**Comment by kaniini** [comment]
I still think the page, in it's entirety, is inappropriate, and do not want my mechanism documented there.  Thank you for respecting my request.

**Pull request closed without merging**